pkg/adaptor/mongodb/mongodb.go

```diff
@@ -127,7 +148,10 @@ func (m *MongoDB) SampleConfig() string {
 
 // Connect tests the mongodb connection and initializes the mongo session
 func (m *MongoDB) Connect() error {
-	_, err := m.client.Connect()
+	s, err := m.client.Connect()
+	if s, ok := s.(client.Closer); ok {
+		s.Close()
+	}
```

Comment:
we were leaking a connection here which meant there was a goroutine that would run forever when used as a library, should fix https://github.com/compose/transporter/issues/265